docs/DEBUGGING.md

```diff
@@ -162,6 +162,12 @@ PID   USER     TIME  COMMAND
 ~ #
 ```
 
+## Prevent shutdown of a container applications
+
+If container's entrypoint (e.g. `init` script) misbehaves and exits, it becomes very difficult to debug such a container, because hosting Vm shuts down immediately by explicit poweroff call.
+
+If EVE_CONTAINER_NO_POWEROFF=1 environment variable is set on the controller side, then the hosting Vm does not shut down, but waits for further debugging session, for example attaching to a `prime-cons` console.
```

Comment:
Maybe this should be more specific and say something like "is set in the cloud-init userdata on the controller side"?

Reply:
Maybe, but on the controller there is no term "cloud-init", there is "Add Custom Config Template". So please suggest here what proper term to use.

Reply:
In the EVE API we have user_data with this comment
  // App Instance initialization configuration data provided by user
  // This will be used as "user-data" in cloud-init
  // Empty string will indicate that cloud-init is not required
  // It is also used to carry environment variables for containers.
  // XXX will be deprecated and replaced by the cipherData below.
  string userData = 11;

So suggested text would be "set by the controller in the app instance userData/cipherData".

Reply:
cool, will do.